src/miner_util.erl

```diff
@@ -144,6 +145,10 @@ random_miner_predicate(Peer) ->
     not libp2p_peer:is_stale(Peer, timer:minutes(360)) andalso
         maps:get(<<"release_info">>, libp2p_peer:signed_metadata(Peer), undefined) /= undefined.
 
+random_non_miner_predicate(Peer) ->
+    not libp2p_peer:is_stale(Peer, timer:minutes(360)) andalso
+        maps:get(<<"release_info">>, libp2p_peer:signed_metadata(Peer), undefined) == undefined.
```

Comment:
guessing this only gets set in the validator release so a miner is always going to return the default value supplied